app/views/forms/odk/_form_controls_node.xml.erb

```diff
@@ -0,0 +1,48 @@
+<% if node.childless? %>
```

Comment:
This file is really really complicated because it handles any combo of: nesting, repeat, multilevel, grid, hidden. It is fragile and difficult to debug. Not sure how to make it better.